data_types/src/lib.rs

```diff
@@ -377,32 +377,32 @@ impl std::fmt::Display for ParquetFileId {
     }
 }
 
-/// Data object for a kafka topic
+/// Data object for a kafka topic.
```

Comment:
Several types in this crate are documented as "Data object for a..." I think higher-level descriptions of these concepts would be better here. For example:

```suggestion
/// A Kafka topic. Kafka topics are used to...
```